src/DataStructures/Mesh.cpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/Mesh.hpp"
+
+#include <pup.h>  // IWYU pragma: keep
+
+#include "Parallel/PupStlCpp11.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+template <size_t Dim>
+template <size_t N, Requires<(N > 0)>>
```

Comment:
is the size_t N used here? At first glance it seems like it should be Dim which is > 0 and not N

Reply:
This is how it is implemented in `Index.hpp`, could it be done differently? We want `Mesh` (and `Index`) to implement `slice_away` only for `Dim > 0`, so `N` is introduced to enforce this condition.

Reply:
Use `Requires<(N > 0 and N==Dim)>` as well as in the hpp